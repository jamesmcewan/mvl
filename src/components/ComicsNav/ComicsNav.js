import React, { Component } from 'react';
import styled from 'styled-components';
// import Link from 'react-router-dom';
import { Comics } from '..';
import getData from '../../getData';
import Loading from '../Loading/Loading';

const ComicsGrid = styled.div`
  display: flex;
  align-items: flex-start;
  flex-direction: column;

  @media screen and (min-width: 700px) {
    padding-top: 170px;
  }
`;

export default class ComicsNav extends Component {
  static defaultProps = {
    match: {
      url: '/'
    }
  }

  state = {
    comics: [],
    isSpotlightVisible: false,
    comicsLoading: true
  }

  getMarvelData = dateDescriptor => {
    getData(dateDescriptor)
      .then(res => {
        const { comics } = res;
        return this.setState({
          comics,
          comicsLoading: false
        });
      })
      .catch(err => console.log(err));
  };

  resetComics = () => {
    return this.setState({
      comics: [],
      comicsLoading: true
    })
  }

  componentWillMount() {
    const { match: { url }} = this.props;
    this.resetComics();
    if (url === '/') {
      this.getMarvelData('/');
      return;
    }

    this.getMarvelData(url);
  }


  render() {
    const { comics, comicsLoading } = this.state;

    return (
      <ComicsGrid>
        {comicsLoading && <Loading /> }
        {!comicsLoading && <Comics {...{ comics }} /> }
      </ComicsGrid>
    )
  }
}