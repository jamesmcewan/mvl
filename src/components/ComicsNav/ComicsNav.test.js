import React from 'react';
import { mount } from 'enzyme';
import { ComicsNav } from './ComicsNav';
import 'jest-styled-components';

describe('ComicsNav', () => {
  let getComics;
  beforeEach(() => {
    getComics = jest.fn();
  });
  it('should render without crashing', () => {
    const wrapper = mount(<ComicsNav {...{ getComics }} />);
    expect(wrapper).toHaveLength(1);
  });

  it('should render the correct JSX', () => {
    const wrapper = mount(<ComicsNav {...{ getComics }} />);
    expect(wrapper).toMatchSnapshot();
  });
});
