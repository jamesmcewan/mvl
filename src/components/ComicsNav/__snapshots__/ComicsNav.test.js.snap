// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicsNav should render the correct JSX 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin: 52px auto;
}

.c2 {
  -webkit-animation: iVXCSc 2s linear infinite;
  animation: iVXCSc 2s linear infinite;
  font-size: 100px;
  width: 100px;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media screen and (min-width:700px) {
  .c0 {
    padding-top: 170px;
  }
}

<ComicsNav
  getComics={[Function]}
>
  <styled.div>
    <div
      className="c0"
    >
      <Loading>
        <styled.div>
          <div
            className="c1"
          >
            <styled.div>
              <div
                className="c2"
              >
                âš™
              </div>
            </styled.div>
          </div>
        </styled.div>
      </Loading>
    </div>
  </styled.div>
</ComicsNav>
`;
